### ISSUE TITLE
[DOC] - Login do usuário

### ISSUE THREAD

**Issue by aleodoni** [CMCuritiba/processo_eletronico#17]
## Documentação sobre o login do usuário no sistema

Caminho 1: As informações usadas no login serão buscadas do LDAP

-usuario
-email
-nome_usuário (nome completo)
-setor_usuário (id do setor)
-area_usuario (id da área)
-nome_setor_usuario (nome do setor)
-nome_area_usuario (nome da área)

Prós:
- Não dependerá mais da elotech
- O código fica mais fácil, somente uma consulta no LDAP
- Não interessa pro sistema quem vai atualizar estas informações

Contras:
- Quem vai construir essa solução?
- Como vamos testar isso agora em desenvolvimento?
- Quanto tempo para isso ser desenvolvido?
- Como vamos atualizar isso?
- Como vou usar isso aqui remotamente? Vou ter que rodar um docker aqui? Não conseguimos nem um acesso remoto ao bd...

------------------------------------------------

Caminho 2: As informações serão parte buscadas do LDAP e parte de tabelas próprias do sistema.

-usuario(LDAP)
-email(LDAP)
-nome_usuário (nome completo)(LDAP)
-setor_usuário (id do setor)(sistema)
-area_usuario (id da área)(sistema)
-nome_setor_usuario (nome do setor)(sistema)
-nome_area_usuario (nome da área)(sistema)

Funciona da seguinte maneira:
1) Foi criada uma tabela de setores no sistema, esta tabela tem a seguinte estrutura:

set_nome character varying(200), -- Nome do setor
set_sigla character varying(100), -- Sigla do setor
set_inativacao date, -- Data de inativação, se for null é ativa
set_id integer NOT NULL, -- Código do setor, chave da tabela
set_id_area integer, -- id da área que pertence o setor
set_ativo boolean, -- se o setor está ativo
set_tipo character(1), -- tipo do setor: N-> normal , G-> gabinete, E-> especial

2) Foi criada uma tabela de lotacao, esta tabela tem a seguinte estrutura:

matricula character varying(5) NOT NULL, --matricula da pessoa
pes_nome character varying(200), -- nome da pessoa
set_id integer, -- id do setor, ligado com o set_id da tabela setor
pes_login character varying(100), -- login da pessoa

3) Com essas tabelas preenchidas criei a seguinte view:

SELECT a.set_id,
a.set_nome AS nome_setor,
a.set_sigla AS nome_sigla,
b.matricula,
b.pes_nome AS nome,
b.pes_login AS login,
area.set_id AS set_id_area,
area.set_nome AS nome_area
FROM spa2.setor a,
spa2.lotacao b,
spa2.setor area
WHERE a.set_id = b.set_id AND a.set_id_area = area.set_id AND a.set_ativo = true;

4) Faço o login no LDAP, se achou a pessoa lá retorna os dados do LDAP e faz a consulta dos dados restantes pelo login na consulta.

------------------------
Agora vamos aos prós e contras:

Prós:
- Tenho como testar aqui e andar pra frente
- Não dependo da elotech
- Não dependo de implementação de nada no LDAP
- A implementação é simples e já ta pronta aqui

Contras:
- Tenho que cadastrar setor e lotações no sistema( já faço isso hoje no spa, o Celso vem pedir pra mim e em 3 minutos ou menos faço isso)
- Tem que ter uma carga inicial nas tabelas(na de setor já tenho aqui, na de lotações tenho uma de teste, mas não vejo tanto trabalho assim)

----------------------------------------------------

* Lembrando que vamos ter funcionários novos esse ano do concurso e também eleições de vereadores.
* A solução 2 pode ser atualizada criando um serviço a parte que veja o que foi atualizado no LDAP ou no RH e atualize as tabelas, mas isso para um momento futuro.

**Comment by brunosoli** [comment]
Não sei se tem algum porém que não percebi, mas o gerenciamento de usuários e grupos já é todo realizado no LDAP, não faz muito sentido pra mim duplicar as informações e duplicar o trabalho de gerenciamento.

Até onde eu sei não há nenhuma implementação a ser feita no LDAP, as informações já estão lá, não é?

O desenvolvimento e testes fica resolvido com a issue #20.

**Comment by aleodoni** [comment]
Concordo, porém como conversado, teriam que ser criados outros campos do LDAP necessários como área (diretoria que o setor, depto, etc) está subordinado bem como os setores especiais como Comissão Executiva por exemplo. 

Hoje já foi desenvolvida pelo @FabioDerossoTeixeira  uma solução que resolve bem o nosso problema e seguiremos com ela.

Futuramente penso que teremos que desenvolver uma solução para manter sincronizados os dados do LDAP e também as informações do sistema do RH.
